test_matrix.py

```diff
@@ -0,0 +1,78 @@
+import unittest
+from Matrix import Matrix
+
+
+class Test_matrix(unittest.TestCase):
+    def setUp(self):
+        self.x = Matrix(((1, 2), (3, 4)))
+        self.y = Matrix(((4, 3), (2, 1)))
+        self.zeros = Matrix(((0, 0), (0, 0)))
+        self.tens = Matrix(((10, 10), (10, 10)))
+        self.negative = Matrix(((-1, -1), (-1, -1)))
+        self.three = Matrix(((1, 2, 0), (0, 1, 2), (2, 0, 1)))
+        self.half = Matrix(((0.5, 0.5), (0.5, 0.5)))
+
+    def test_init_failed(self):
+        with self.assertRaises(ValueError):
+            (Matrix("hello"))
+        with self.assertRaises(ValueError):
+            (Matrix(((2, 1), (1, 2, 3))))
+
+    def test_add(self):
+        self.assertEqual(self.x + self.y, Matrix(((5, 5), (5, 5))))
+        self.assertEqual(self.x + Matrix(((-2, -3), (-4, -5))), Matrix(((-1, -1), (-1, -1))))
+        self.assertEqual(self.three + self.three, Matrix(((2, 4, 0), (0, 2, 4), (4, 0, 2))))
+
+    def test_add_failed(self):
+        with self.assertRaises(SyntaxError):
+            (Matrix.__add__(self.x, 2))
+
+    def test_mul(self):
+        self.assertEqual(self.zeros * self.x, self.zeros)
+        s = Matrix(((1, 1, 1), (1, 1, 1), (1, 1, 1)))
+        self.assertEqual(s * self.three, Matrix(((3, 3, 3), (3, 3, 3), (3, 3, 3))))
+        self.assertEqual(self.tens * self.half, self.tens)
+        self.assertEqual(self.negative * self.negative, Matrix(((2, 2), (2, 2))))
+        self.assertEqual(self.x * 3, Matrix(((3, 6), (9, 12))))
```

Comment:
what will happen if you do 3 * self.x?

Reply:
Common Michael. I asked you to test that days ago (and you will see that it exposes something missing in your code).